infrastructure/lifecyclehook.yaml

```diff
@@ -0,0 +1,145 @@
+Description: >
+    This template deploys a Lambda Function and Auto Scaling Lifecycle Hook to drain Tasks from your Container Instances when an Instance is selected for Termination in your Auto Scaling Group.
+Parameters:
+  Cluster:
+    Type: String
+    Description: Name of ECS Cluster
+  ECSAutoScalingGroupName:
+    Type: String
+    Description: Name of Auto Scaling Group
+Resources:
+  NotificationTopic:
+    Type: 'AWS::SNS::Topic'
+    Properties:
+      Subscription:
+        - Endpoint: !GetAtt 
+            - LifecycleHandlerFunction
+            - Arn
+          Protocol: lambda
+    DependsOn: LifecycleHandlerFunction
```

Comment:
Is that DependsOn necessary? I think the !GetAtt automatically creates a dependency?